crates/ruff_cli/src/diagnostics.rs

```diff
@@ -343,9 +343,7 @@ pub(crate) fn lint_path(
 
     let notebook_indexes = if let SourceKind::IpyNotebook(notebook) = source_kind {
         FxHashMap::from_iter([(
-            path.to_str()
-                .ok_or_else(|| anyhow!("Unable to parse filename: {:?}", path))?
-                .to_string(),
+            path.to_string_lossy().to_string(),
```

Comment:
This change is to keep the value consistent. We use `.to_string_lossy` everywhere.